### PR TITLE
Added curanet dns support

### PR DESCRIPTION
**Pull request by peterlh**
Hi,

Added curanet dns support using rest api.

### CODE REVIEW

**Comment by Neilpang** [comment]
please add the usage to the wiki page, and create an  issue for reporting bugs.
see: https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide

**Comment by peterlh** [comment]
got it, created issue and modified wiki page
i assumed this where to be done after PR :-)

File: dnsapi/dns_curanet.sh [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    _err "You don't specify curanet api client and secret."
+    _err "Please create your auth info and try again."
+    return 1
+  fi
+
+  #save the credentials to the account conf file.
+  _saveaccountconf_mutable CURANET_AUTHCLIENTID "$CURANET_AUTHCLIENTID"
+  _saveaccountconf_mutable CURANET_AUTHSECRET "$CURANET_AUTHSECRET"
+
+  gettoken
```
Comment (Neilpang) [inline dnsapi/dns_curanet.sh:36]: check the return value of `gottoken`

File: dnsapi/dns_curanet.sh [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+  _get_root "$fulldomain"
```
Comment (Neilpang) [inline dnsapi/dns_curanet.sh:38]: check the return value of `_get_root`

File: dnsapi/dns_curanet.sh (hunk 1 above)
Comment (peterlh) [reply dnsapi/dns_curanet.sh]: done

**Comment by peterlh** [comment]
pushed changes to master branch

**Comment by Neilpang** [comment]
your latest ci checks are not passing:
https://github.com/peterlh/acme.sh/actions/runs/1835053677

**Comment by peterlh** [comment]
passing again

**Comment by peterlh** [comment]
hi, any action need on my part, in order to proceed here?

**Pull request merged**